File: Daily Test/Java/Sort N Dates 4.3.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();

        Map<String,Integer> month = new HashMap<>();
        month.put("JAN",1); month.put("FEB",2); month.put("MAR",3);
        month.put("APR",4); month.put("MAY",5); month.put("JUN",6);
        month.put("JUL",7); month.put("AUG",8); month.put("SEP",9);
        month.put("OCT",10); month.put("NOV",11); month.put("DEC",12);

        List<String[]> dates = new ArrayList<>();

        for(int i=0;i<n;i++) {
            String[] parts = sc.nextLine().split("-");
            Arrays.sort(parts, Comparator.comparingInt(String::length));
            dates.add(parts);
        }

        Collections.sort(dates, (a,b) -> {
            if(Integer.parseInt(a[2]) != Integer.parseInt(b[2]))
                return Integer.parseInt(a[2]) - Integer.parseInt(b[2]);

            if(month.get(a[1].toUpperCase()) != month.get(b[1].toUpperCase()))
                return month.get(a[1].toUpperCase()) -
                       month.get(b[1].toUpperCase());

            return Integer.parseInt(a[0]) - Integer.parseInt(b[0]);
        });

        for(String[] d : dates)
            System.out.println(d[0]+"-"+d[1]+"-"+d[2]);
    }
}

File: Daily Test/Python/Sort N Dates 4.3.2025.py
n = int(input())
dates = []

month_map = {
    'JAN':1,'FEB':2,'MAR':3,'APR':4,'MAY':5,'JUN':6,
    'JUL':7,'AUG':8,'SEP':9,'OCT':10,'NOV':11,'DEC':12
}

for _ in range(n):
    parts = input().strip().split('-')
    parts.sort(key=lambda x: len(x))
    dates.append(parts)

dates.sort(key=lambda x: (int(x[2]), month_map[x[1].upper()], int(x[0])))

for d, m, y in dates:
    print(f"{d}-{m}-{y}")